test/units/cli/test_galaxy.py

```diff
@@ -444,3 +450,168 @@ def test_template_ignore_similar_folder(self):
+                          msg='namespace was not set properly in metadata')
+        self.assertEquals(metadata.get('name', ''), 'delete_me_skeleton', msg='name was not set properly in metadata')
+        self.assertEquals(metadata.get('version', ''), '0.1.0', msg='version was not set properly in metadata')
+        self.assertEquals(metadata.get('authors', []),
+                          ['Ansible Cow <acow@bovineuniversity.edu>', 'Tu Cow <tucow@bovineuniversity.edu>'],
+                          msg='authors was not set properly in metadata')
+
+    def test_empty_files_dir(self):
+        for empty_dir in ['plugins/action', 'plugins/filter', 'plugins/inventory', 'plugins/lookup',
+                          'plugins/module_utils', 'plugins/modules']:
+            empty_dir = os.path.join(self.collection_dir, empty_dir)
+
+            self.assertTrue(os.path.isdir(empty_dir))
+            self.assertListEqual(os.listdir(empty_dir), [],
+                                 msg='we expect the directory to be empty, is ignore working?')
+
+    def test_template_ignore_non_j2_extension(self):
+        test_file = os.path.join(self.collection_dir, 'docs', 'My Collection.md')
+        self.assertTrue(os.path.exists(test_file), msg="The 'docs/My Collection.md' file doesn't seem to exist")
+        with open(test_file, 'r') as f:
+            contents = f.read()
+        expected_contents = 'Welcome to my test collection doc for {{ namespace }}.'
+        self.assertEqual(expected_contents, contents.strip(),
+                         msg="'docs/My Collection.md' does not contain what it should, is it being rendered?")
+
+    def test_template_ignore_jinja(self):
+        for template_dir in ['playbooks/templates', 'playbooks/templates/subfolder',
+                             'roles/common/templates', 'roles/common/templates/subfolder']:
+            test_conf_j2 = os.path.join(self.collection_dir, template_dir, 'test.conf.j2')
+            self.assertTrue(
+                os.path.exists(test_conf_j2),
+                msg="The {0} template doesn't seem to exist, is it being rendered as test.conf?".format(test_conf_j2)
+            )
+
+            with open(test_conf_j2, 'r') as f:
+                contents = f.read()
+            expected_contents = '[defaults]\ntest_key = {{ test_variable }}'
+            self.assertEqual(expected_contents, contents.strip(),
+                             msg="test.conf.j2 doesn't contain what it should, is it being rendered?")
+
+    def test_skeleton_option(self):
+        self.assertEquals(self.collection_skeleton_path, context.CLIARGS['collection_skeleton'],
+                          msg='Skeleton path was not parsed properly from the command line')
+
+    def test_readme(self):
+        readme_path = os.path.join(self.collection_dir, 'README.md')
+        self.assertTrue(os.path.exists(readme_path), msg='Readme doesn\'t exist')
+
+
+def test_invalid_skeleton_path():
+    with pytest.raises(AnsibleError) as err:
```

Comment:
It's cleaner to use matches:
```suggestion
    with pytest.raises(
    		AnsibleError,
    		matches="- the skeleton path '/fake/path' "
    		"does not exist, cannot init collection",
	):
```

Reply:
Thanks I wasn't aware of this, looks like the kwarg is just `match`.

Reply:
Oh, right. it's just `match=`